library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationSource.java

```diff
@@ -89,17 +89,23 @@ void startSettingsApiResolutionForResult(Status status, Activity activity) throw
 
     @SuppressWarnings("ResourceType")
     void requestLocationUpdate() {
-        LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        if (isGoogleApiClientConnected()) {
+            LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        } else {
+            connectGoogleApiClient();
+            requestLocationUpdate();
+        }
     }
 
     void removeLocationUpdates() {
-        LocationServices.FusedLocationApi.removeLocationUpdates(googleApiClient, this);
+        if (isGoogleApiClientConnected())
```

Comment:
I will change other things. As I said I am new in this kind of project. Can you please explain what I need to do except these code changes.  Thank you for your time.